src/pages/user/UserInstitution.tsx

```diff
@@ -72,89 +72,121 @@ const UserInstitution: FC = () => {
     setOpen(!open);
   };
 
-  const updateAuthorityAndDispatch = async (id: string, scn: string) => {
-    const updatedAuthority = await updateInstitutionForAuthority(id, scn);
+  const addOrgunitIdToAuthorityAndDispatch = async (id: string, scn: string) => {
+    const updatedAuthority = await addQualifierIdForAuthority(scn, AuthorityQualifiers.ORGUNIT_ID, id);
     if (updatedAuthority.error) {
       dispatch(setNotification(updatedAuthority.error, NotificationVariant.Error));
     } else if (updatedAuthority) {
       dispatch(setAuthorityData(updatedAuthority));
     }
   };
 
-  const handleAddInstitution = async ({ name, id, subunits }: InstitutionUnit) => {
+  const handleAddInstitution = async ({ name, id, subunits, unit }: FormikInstitutionUnit) => {
     try {
       if (subunits.length === 0) {
-        await updateAuthorityAndDispatch(id, user.authority.systemControlNumber);
+        await addOrgunitIdToAuthorityAndDispatch(id, user.authority.systemControlNumber);
       } else {
         const lastSubunit = subunits.slice(-1)[0];
-        await updateAuthorityAndDispatch(lastSubunit.id, user.authority.systemControlNumber);
+        await addOrgunitIdToAuthorityAndDispatch(lastSubunit.id, user.authority.systemControlNumber);
       }
     } catch (error) {
       dispatch(setNotification(t('feedback:error.update_authority'), NotificationVariant.Error));
     }
     // TODO: remove this when we get data from backend
     const filteredSubunits = subunits.filter((subunit: InstitutionUnitBase) => subunit.name !== '');
-    setUnits([...units, { id, name, subunits: filteredSubunits }]);
+    setUnits([...units, { id, name, subunits: filteredSubunits, unit }]);
+  };
 
+  const onSubmit = async (values: FormikInstitutionUnit, { resetForm }: any) => {
+    if (editMode && values.editId) {
+      const organizationUnitId = values.subunits.length > 0 ? values.subunits.slice(-1)[0].id : values.id;
```

Comment:
`.pop()`? :)
slice() er vannskli for mei

Reply:
Pop fjerner elementet fra lista, mens slice ikke gjør det. Men siden vi ikke bruker values.subunits videre her, så kan jeg bytte til pop 👍

Det blir et `!` her da, pga https://www.bennadel.com/blog/3582-using-the-non-null-assertion-operator-to-fix-shift-and-pop-errors-in-typescript-3.htm